source/main/modules/window-manager/index.ts

```diff
@@ -219,6 +235,64 @@ export default class WindowManager {
       return
     }
 
+    this._mainWindow.on('show', () => {
+      const leaveAppRunning = Boolean(global.config.get('system.leaveAppRunning'))
+      if (leaveAppRunning) {
+        if (process.platform === 'darwin') {
+          if (this._tray == null) {
+            let basepath = app.getAppPath()
+            this._tray = new Tray(path.join(__dirname, './assets/icons/128x128.png'))
+
+            const contextMenu = Menu.buildFromTemplate([
+              {
+                label: 'Show Zettlr',
+                click: () => {
+                  this.showAnyWindow()
+                },
+                type: 'normal'
+              },
+              { label: '', type: 'separator' },
+              {
+                label: 'Quit',
+                click: () => {
+                  app.quit()
```

Comment:
The tray isn't working for me so I can't test at the moment, but does `app.quit()` follow this part of the [tray design](https://github.com/UofA-SPI21-team7/zettlr-assignment/wiki/Tray-design)?:

> If there are unsaved changes, show and bring-to-the-front the Zettlr window and display the 'There are unsaved changes' modal dialog.

 I tried to follow the  Zettlr code and if I follow correctly this will quit immediately?

Reply:
yes, the app.quit() bind to 'will-quit' event. In fact, when zettlr fired this event will also trigger the un-saved content check

Reply:
I create an icon with nativeImage, but you do not allow me to do and want me to create like this:

this._tray = new Tray(path.join(__dirname, './assets/icons/22x22_white.png'))

In fact, this way is not the correct way to create a tray icon for macOS. But you insist on your way. So I have no idea about this point.